mm2src/mm2_main/src/lp_ordermatch.rs

```
@@ -310,50 +312,202 @@ fn process_trie_delta(
+        // The message was processed, return Ok to forward it
         None => return Ok(()),
     };
 
-    let response = request_one_peer::<SyncPubkeyOrderbookStateRes>(
-        ctx.clone(),
-        P2PRequest::Ordermatch(req),
-        propagated_from_peer.clone(),
-    )
-    .await
-    .map_mm_err()?
-    .ok_or_else(|| {
-        MmError::new(OrderbookP2PHandlerError::P2PRequestError(format!(
-            "No response was received from peer {propagated_from_peer} for SyncPubkeyOrderbookState request!"
-        )))
-    })?;
+    // Phase 2: Prepare expected roots and the initial set of pairs to sync.
+    let OrdermatchRequest::SyncPubkeyOrderbookState {
+        trie_roots: expected_roots_by_pair,
+        ..
+    } = req
+    else {
+        error!(
+            "`process_keep_alive` returned unexpected `OrdermatchRequest` variant: {:?}",
+            req
+        );
+        return Ok(());
+    };
+    let mut remaining_pairs: HashSet<AlbOrderedOrderbookPair> = expected_roots_by_pair.keys().cloned().collect();
+
+    // Build the peer sequence once: origin first, then current mesh (excluding origin).
+    let peers = {
+        let p2p_cmd_tx = P2PContext::fetch_from_mm_arc(&ctx).cmd_tx.lock().clone();
+        let mut mesh = get_relay_mesh(p2p_cmd_tx).await;
+        mesh.retain(|p| p != &propagated_from);
+        std::iter::once(propagated_from.clone())
+            .chain(mesh.into_iter())
+            .collect::<Vec<_>>()
+    };
+
+    // Phase 3: Query peers sequentially until all pairs are synced, or we run out of peers.
+    let mut consulted = Vec::with_capacity(peers.len());
+    for peer in peers {
+        if remaining_pairs.is_empty() {
+            break;
+        }
+        request_and_apply_pubkey_state_sync_from_peer(
+            &ctx,
+            &ordermatch_ctx.orderbook,
+            &from_pubkey,
+            &peer,
+            &expected_roots_by_pair,
+            &mut remaining_pairs,
+        )
+        .await?;
```

what about ignoring the error? i guess we wanna perform these requests in a best-effort basis. if we fail to get a response (idk how fail cases happen since `PeerDecodedResponse` is not really well-documented :/), we can try the other peers we have in the list.

---

Will not ignore this error in the next commit

---

Nevermind, will ignore it as if any error is returned from `request_and_apply_pubkey_state_sync_from_peer` `remaining_pairs` will not be empty and it will be a `SyncFailure`